File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        int res[] = new int[2];
        HashMap<Integer,Integer> mp = new HashMap<>();
        
        for(int i=0;i<nums.length;i++){
            
            if(mp.containsKey(target-nums[i])){
                res[0] =i;
                res[1] = mp.get(target-nums[i]);
                
                return res;
            }
            mp.put(nums[i], i);
            
            
        }
        return res;
    }
}

File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    res := make([]int, 2)
    mp := make(map[int]int)
    
    for i:= 0; i< len(nums);i++{
        cp := target - nums[i]
        
        if index, found := mp[cp]; found{
            res[0] =i
            res[1] = index
            return res
        }
        mp[nums[i]] = i
    }
    return res
}